packages/flutter_tools/lib/src/windows/topaz_ofd_validator.dart

```diff
@@ -0,0 +1,58 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:process/process.dart';
+
+import '../base/io.dart';
+import '../doctor_validator.dart';
+
+const String kCoreProcessPattern = r'Topaz OFD\\Warsaw\\core\.exe';
+
+class TopazOfdValidator extends DoctorValidator {
+  const TopazOfdValidator({required this.processLister}) : super('Topaz OFD');
+
+  final ProcessLister processLister;
+
+  @override
+  Future<ValidationResult> validate() async {
+    final ProcessResult processResult = await processLister.getProcessesWithPath('core');
+    final int exitCode = processResult.exitCode;
+    if (exitCode != 0) {
+      return ValidationResult(
+        ValidationType.missing,
+        <ValidationMessage>[
+          ValidationMessage.hint('Check for Topaz OFD did not complete normally. Returned exit code $exitCode'),
+        ],
+        statusInfo: 'Get-Process failed to complete',
+      );
+    }
+    final String tasks = processResult.stdout as String;
+    final RegExp pattern = RegExp(kCoreProcessPattern, multiLine: true, caseSensitive: false);
+    final bool matches = pattern.hasMatch(tasks);
+    if (matches) {
+      return const ValidationResult(
+        ValidationType.missing,
+        <ValidationMessage>[
+          ValidationMessage.hint('The Topaz OFD Security Module process has been found running. If you are unable to build, you will need to disable it.'),
+        ],
+        statusInfo: 'Topaz OFD may be running');
+    } else {
+      return const ValidationResult(
+        ValidationType.success,
+        <ValidationMessage>[],
+      );
+    }
+  }
+}
+
+class ProcessLister {
+  ProcessLister(this.processManager);
+
+  final ProcessManager processManager;
+
+  Future<ProcessResult> getProcessesWithPath(String? filter) async {
+    final String argument = filter != null ? 'Get-Process $filter | Format-List Path' : 'Get-Process | Format-List Path';
```

Comment:
Could we filter using Dart logic instead and remove the `$filter` argument?

This could be susceptible to command injection attacks if we ever accepted user-controlled input. The code today doesn't seem affected, but I'd avoid this kind of concern altogether if it's not necessary.

Reply:
Also, the filter breaks if your machine does not have a process that matches the filter:

![image](https://github.com/flutter/flutter/assets/737941/0f47e56a-8c61-42da-b78b-e4ae5f64a08a)

Standard error:

```
Get-Process : Cannot find a process with the name "core". Verify the process name and call the cmdlet again.
At line:1 char:1
+ Get-Process core
+ ~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (core:String) [Get-Process], ProcessCommandException
    + FullyQualifiedErrorId : NoProcessFoundForGivenName,Microsoft.PowerShell.Commands.GetProcessCommand
```